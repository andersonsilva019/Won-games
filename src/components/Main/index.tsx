import * as S from './styles'

export default function Main({
  title = 'Nextjs',
  description = 'Typescript, Reactjs, Nextjs e Styled components'
}) {
  return (
    <S.Container>
      <S.Logo
        src="/images/logo.svg"
        alt="Imagem de um átomo e React Avançado escrito ao lado"
      />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration
        src="/images/hero-illustration.svg"
        alt="Um desenvolvedor de frente para uma tela com código"
      />
    </S.Container>
  )
}
