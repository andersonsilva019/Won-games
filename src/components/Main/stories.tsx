import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main
} as Meta

export const Basic: Story = args => <Main {...args} />

Basic.args = {
  title: 'Title default',
  description: 'Description default'
}

export const Secondary: Story = args => <Main {...args} />

Secondary.args = {
  title: 'Title default Secondary',
  description: 'Description  Secondary'
}
