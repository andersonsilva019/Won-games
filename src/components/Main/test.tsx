import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)
    expect(screen.getByRole('heading', { name: /nextjs/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the styles correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b',
      color: '#fff',
      width: '100%',
      height: '100%',
      padding: '3rem',
      display: 'flex',
      'flex-direction': 'column',
      'text-align': 'center',
      'align-items': 'center',
      'justify-content': 'center'
    })
  })
})
