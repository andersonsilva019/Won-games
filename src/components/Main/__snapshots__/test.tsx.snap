// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Container-sc-1cs6c8-0 gzogom"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado"
    class="styles__Logo-sc-1cs6c8-1 igVxLV"
    src="/images/logo.svg"
  />
  <h1
    class="styles__Title-sc-1cs6c8-2 VHROD"
  >
    Nextjs
  </h1>
  <p
    class="styles__Description-sc-1cs6c8-3 ujdfM"
  >
    Typescript, Reactjs, Nextjs e Styled components
  </p>
  <img
    alt="Um desenvolvedor de frente para uma tela com código"
    class="styles__Illustration-sc-1cs6c8-4 frJyYJ"
    src="/images/hero-illustration.svg"
  />
</main>
`;
